### ISSUE TITLE
Crash due to infinite recursion

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3399]
The pylance telemetry is showing a new (previously unseen) crash with the following stack trace:

```
RangeError: Maximum call stack size exceeded
    at getClassMemberIterator (<REDACTED: user-file-path>:1029:18)
    at lookUpClassMember (<REDACTED: user-file-path>:1016:23)
    at getBoundMethod (<REDACTED: user-file-path>:1851:28)
    at canAssignType (<REDACTED: user-file-path>:21522:37)
    at partiallySpecializeFunctionForBoundClassOrObject (<REDACTED: user-file-path>:23989:18)
    at bindFunctionToClassOrObject (<REDACTED: user-file-path>:23846:24)
    at getBoundMethod (<REDACTED: user-file-path>:1856:37)
    at canAssignType (<REDACTED: user-file-path>:21522:37)
    at partiallySpecializeFunctionForBoundClassOrObject (<REDACTED: user-file-path>:23989:18)
    at bindFunctionToClassOrObject (<REDACTED: user-file-path>:23846:24)
    at getBoundMethod (<REDACTED: user-file-path>:1856:37)
    at canAssignType (<REDACTED: user-file-path>:21522:37)
    at partiallySpecializeFunctionForBoundClassOrObject (<REDACTED: user-file-path>:23989:18)
    at bindFunctionToClassOrObject (<REDACTED: user-file-path>:23846:24)
    at getBoundMethod (<REDACTED: user-file-path>:1856:37)
    at canAssignType (<REDACTED: user-file-path>:21522:37)
    at partiallySpecializeFunctionForBoundClassOrObject (<REDACTED: user-file-path>:23989:18)
    at bindFunctionToClassOrObject (<REDACTED: user-file-path>:23846:24)
    at getBoundMethod (<REDACTED: user-file-path>:1856:37)
    at canAssignType (<REDACTED: user-file-path>:21522:37)
    at partiallySpecializeFunctionForBoundClassOrObject (<REDACTED: user-file-path>:23989:18)
    at bindFunctionToClassOrObject (<REDACTED: user-file-path>:23846:24)
    at getBoundMethod (<REDACTED: user-file-path>:1856:37)
    at canAssignType (<REDACTED: user-file-path>:21522:37)
    at partiallySpecializeFunctionForBoundClassOrObject (<REDACTED: user-file-path>:23989:18)
    at bindFunctionToClassOrObject (<REDACTED: user-file-path>:23846:24)
    at getBoundMethod (<REDACTED: user-file-path>:1856:37)
    at canAssignType (<REDACTED: user-file-path>:21522:37)
    at partiallySpecializeFunctionForBoundClassOrObject (<REDACTED: user-file-path>:23989:18)
    at bindFunctionToClassOrObject (<REDACTED: user-file-path>:23846:24)
    at getBoundMethod (<REDACTED: user-file-path>:1856:37)
    at fs (<REDACTED: user-file-path>:21522:37)
```

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.242, which I just published. It will also be included in the next release of pylance.
